[fix] Change FileManager class method and update its interface

StylishPanther opened the pull request:
# Back ground
Architecture is changed, so update filemanager
with changing method parameter and create its interface.

Also, as method is changed, unit test is also updated.

# Test Result

Test result is as below:
```
Running main() from C:\Users\User\source\repos\SSDTestShell\packages\gmock.1.11.0\lib\native\src\gtest\src\gtest_main.cc
[==========] Running 7 tests from 2 test suites.
[----------] Global test environment set-up.
[----------] 6 tests from FileManagerTest
[ RUN      ] FileManagerTest.file_manager_test_init
[       OK ] FileManagerTest.file_manager_test_init (0 ms)
[ RUN      ] FileManagerTest.file_manager_test_open
[       OK ] FileManagerTest.file_manager_test_open (0 ms)
[ RUN      ] FileManagerTest.file_manager_test_close
[       OK ] FileManagerTest.file_manager_test_close (0 ms)
[ RUN      ] FileManagerTest.file_manager_test_read
[       OK ] FileManagerTest.file_manager_test_read (0 ms)
[ RUN      ] FileManagerTest.file_manager_test_write
[       OK ] FileManagerTest.file_manager_test_write (0 ms)
[ RUN      ] FileManagerTest.file_manager_test_write_during_read
[       OK ] FileManagerTest.file_manager_test_write_during_read (0 ms)
[----------] 6 tests from FileManagerTest (2 ms total)
```

StylishPanther committed:
[fix] Change FileManager class method and update its interface

Architecture is changed, so update filemanager
with changing method parameter and create its interface.

Also, as method is changed, unit test is also updated.

wonmyth reviewed (approved):
FileManagerInterface 구현 감사합니다.
리뷰 완료했습니다.

bsjn16sw commented on SSDManager/FileManager.cpp:
```diff
@@ -7,26 +7,28 @@
 
-bool FileManager::write(std::fstream* fs, int index, unsigned int value) {
-    return true;
+unsigned int FileManager::read(std::string name, int index) {
```
read() 한 결과를 `string` 에서 `unsigned int` 로 변환하지 않고 바로 `stfing` 으로 리턴하면 문제가 있을까요?? SSDManager에서도 `value` 값을 `string` 으로 SSDWriter에게 넘겨주기로 하여 질문 드립니다~ 😊

StylishPanther replied:
오프라인에서 이야기 한대로 string으로 변환하도록 하겠습니다! 지금 브랜치 관리를 위해서 이 커밋은 넣고 다음 커밋 때 반영할게요. 리뷰 감사합니다!

StylishPanther replied:
다음 커밋에서 반영되었습니다! 리뷰 감사합니다.

Kanaish commented on SSDManagerTest/FileManagerTest.cpp:
```diff
@@ -3,42 +3,44 @@
 #include "gmock/gmock.h"
 #include "../SSDManager/FileManager.cpp"
+#include "../SSDManager/FileManagerInterface.h"
+#include <string>
```
include string 을 위로 올려서 standard 를 위로 올리면 더 보기가 좋을 것 같습니다.

StylishPanther replied:
../SSDManager/FileManager.cpp 에 include 되어있어서 삭제했습니다! 리뷰 감사합니다!

Kanaish reviewed (changes requested):
구조 변환 확인 했습니다. 수고하셨습니다.
include 순서에 대한 코멘트 확인 부탁드립니다.

bsjn16sw commented on SSDManager/FileManager.h:
```diff
@@ -1,15 +1,22 @@
-    bool write(std::fstream* fs, unsigned int value);
-    bool close(std::fstream* fs);
+    bool init(std::string name);
+    bool open(std::string name);
```
`name` 이 파일명일 것 같은데 `file_name` 은 어떨까요~? 😀

StylishPanther replied:
다음 리팩토링 단계에서 수정하겠습니다!

bsjn16sw commented on SSDManager/FileManager.cpp:
```diff
@@ -7,26 +7,28 @@
 }
 
-bool FileManager::write(std::fstream* fs, unsigned int value) {
+bool FileManager::write(std::string name, int index, unsigned int value) {
```
write() 에서도 `unsigned int` 대신 `string` 으로 통일하면 좋을 것 같습니다!

StylishPanther replied:
다음 커밋에서 반영되었습니다! 리뷰 감사합니다.

bsjn16sw reviewed (approved):
고생 많으셨습니다! 코멘트 몇 개 남겨두었습니다!

jiyeon-08 commented:
수고많으셨습니다! cpplint rule(tab->2 space) 적용 안된 것 같은데 확인 부탁드립니다~!!

miryo9yd reviewed (approved):
Code Review 하였습니다.

StylishPanther committed:
[feat] Change method template and implement write method

1) write, read parameter is changed based on string
2) nand write and its unit test is implemented

StylishPanther committed:
[refactor] Change header position

1) string header position is changed

Kanaish reviewed (approved):
수정 확인 완료했습니다. 감사합니다
